src/SSHLibrary/abstractclient.py

```diff
@@ -433,7 +438,11 @@ def read_char(self):
                 server_output += self.shell.read_byte()
                 return self._decode(server_output)
             except UnicodeDecodeError:
-                pass
+                if self.config.handle_decode_errors.upper() == 'STRICT':
+                    self.shell.read() # clear shell to eliminate the characters that cause the error
+                    raise
```

Comment:
This does not seem right. Before the idea was to read bytes until they decode to an unicode character. Now this raises error if it first byte does not. Let's say next incoming character is `'å'`, i.e. `b'\xc3\xa5'` as bytes. Now "strict" will be unable to read this character as it fails when attempting decode after first byte.
```python
>>> b'\xc3'.decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 0: unexpected end of data
>>> b'\xc3\xa5'.decode()
'å'
>>>
```

Reply:
This has been fixed in SSHLibrary 3.7.0rc2, see the commit https://github.com/robotframework/SSHLibrary/commit/af3faca5a0ff7ff8c31c4294f575d0be775cf19d.